checks.d/mysql.py

```diff
@@ -810,9 +820,28 @@ def _get_replica_stats(self, db):
         with closing(db.cursor()) as cursor:
             cursor.execute("SHOW SLAVE STATUS;")
             replica_results = dict(cursor.fetchall())
+            cursor.execute("SHOW MASTER STATUS;")
+            binlog_results = dict(cursor.fetchall())
+            if binlog_results:
+                replica_results.update({'Binlog_enabled': True})
 
             return replica_results
 
+    def _get_slave_status(self, db, nonblocking=False):
+        with closing(db.cursor()) as cursor:
+            # querying threads instead of PROCESSLIST to avoid mutex impact on
+            # performance.
+            if nonblocking:
+                cursor.execute("SELECT THREAD_ID, NAME FROM performance_schema.threads WHERE NAME LIKE '%worker'")
+            else:
+                cursor.execute("SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST WHERE COMMAND LIKE '%Binlog dump%'")
```

Comment:
The `slaves` should be handled successfully by the `get_replica_stats()`, because `SHOW SLAVE STATUS` should definitely come up non-empty in a working slave. In `get_replica_stats()` we also go over `SHOW MASTER STATUS`, which will let us know if binary logging is up and running (normally a sign of replication being OK). The final `_get_slave_status()`, is just one last failsafe check we can come up with to check for replication. I will have to put it in a try...except block so that if the user isn't configured with `PROCESS` credentials we don't crash and burn - your previous comment just made me realize this.